src/webgpu/shader/execution/expression/unary/abstract_float_assignment.spec.ts

```diff
@@ -0,0 +1,46 @@
+export const description = `
+Execution Tests for assignment of AbstractFloats to concrete data types
+`;
+
+import { makeTestGroup } from '../../../../../common/framework/test_group.js';
+import { GPUTest } from '../../../../gpu_test.js';
+import { f64, TypeF32, TypeF64 } from '../../../../util/conversion.js';
+import { FP } from '../../../../util/floating_point.js';
+import { f32FilteredF64Range } from '../../../../util/math.js';
+import { makeCaseCache } from '../case_cache.js';
+import { allInputSources, run } from '../expression.js';
+
+import { assignment } from './unary.js';
+
+export const g = makeTestGroup(GPUTest);
+
+export const d = makeCaseCache('unary/abstract_float_assignment', {
+  f32: () => {
+    return f32FilteredF64Range().map(f => {
+      return { input: f64(f), expected: FP.f32.correctlyRoundedInterval(f) };
+    });
+  },
+});
+
+g.test('f32')
+  .specURL('https://www.w3.org/TR/WGSL/#floating-point-conversion')
+  .desc(
+    `
+narrowing down to f32
```

Comment:
So if the RHS does not first explicitly convert the abstract-float to a `f32`, then this is testing [`concretization`](https://www.w3.org/TR/WGSL/#concretization).